pilot/pkg/networking/plugin/plugin.go

```diff
@@ -46,15 +50,26 @@ const (
 	Mixer = "mixer"
 )
 
-// ModelProtocolToListenerProtocol converts from a config.Instance to its corresponding plugin.ListenerProtocol
-func ModelProtocolToListenerProtocol(p protocol.Instance) ListenerProtocol {
+// ModelProtocolToListenerProtocol converts from a config.Protocol to its corresponding plugin.ListenerProtocol
+func ModelProtocolToListenerProtocol(node *model.Proxy, p protocol.Instance) ListenerProtocol {
+	// If protocol sniffing is not enabled, the default value is TCP
+	if !util.IsProtocolSniffingEnabled(node) && p == protocol.Unsupported {
```

Comment:
why do we need to do this?  The old code would have returned unknown in this case.

Reply:
kube/conversion defaults all unknown to TCP before I made the change. Currently kube/conversion could output unsupported protocol. e.g., if users define port name as `test`, before they will get TCP and now they will get Unsupported.